diff --git a/test/CMakeLists.txt b/test/CMakeLists.txt
index cd7ba177..db4d91ec 100644
--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -158,7 +158,7 @@ if (FMT_PEDANTIC
     "-DCMAKE_CXX_FLAGS=${CMAKE_CXX_FLAGS}"
     "-DCMAKE_CXX_STANDARD=${CMAKE_CXX_STANDARD}"
     "-DCXX_STANDARD_FLAG=${CXX_STANDARD_FLAG}"
-    "-DFMT_DIR=${CMAKE_SOURCE_DIR}")
+    "-DFMT_DIR=${YOSYS_SOURCE_DIR}")
 
   # Test if the targets are found from the build directory.
   add_test(
